src/app/_services/api.service.spec.ts

```diff
@@ -0,0 +1,47 @@
+import { TestBed } from '@angular/core/testing';
+import { Injectable } from '@angular/core';
+
+import { HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+
+import { HttpClient, HttpHeaders, HttpErrorResponse } from '@angular/common/http';
+
+import { ApiService } from './api.service';
+
+describe('ApiService', () => {
+
+  let service: ApiService;
+  let httpClient: HttpClient;
+  let httpTestingController: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [ HttpClientTestingModule ],
+      providers: [ ApiService ]
+    });
+
+    // Inject the http service and test controller for each test
+    httpClient = TestBed.get(HttpClient);
+    httpTestingController = TestBed.get(HttpTestingController);
+    service = new ApiService(httpClient);
+  });
+
+  afterEach(() => {
+    // After every test, assert that there are no more pending requests.
+    httpTestingController.verify();
+  });
+
+  it('#getHttpHeaders should return httpHeaders json object ', () => {
+    expect(service.getHttpHeaders()).toBeDefined();
+  });
+
+  it('#getHttpHeadersWithBase64Authorization should return httpHeaders json object ', () => {
+    const paramUsername = "username";
```

Comment:
The test checks whether the object is defined, but not that it is a "httpHeaders json object" as the test name indicates.. so, either the title should be changed to match the test, or the test should do what the title says (check for conditions which would indicate an "httpHeaders json object").